.github/workflows/ngen-conf.yml

```diff
@@ -24,18 +24,34 @@ jobs:
       uses: actions/setup-python@v2
       with:
         python-version: ${{ matrix.python-version }}
-    - name: Install dependencies
-      run: |
-        python -m pip install --upgrade pip
-        if [ -f requirements.txt ]; then pip install -r requirements.txt; fi
-        if [ -f python/requirements.txt ]; then pip install -r python/requirements.txt; fi
-        pip install python/ngen_conf
     - name: Lint with flake8
       run: |
+        pip install -U pip
+        pip install flake8
         # stop the build if there are Python syntax errors or undefined names
         flake8 ./python --count --select=E9,F63,F7,F82 --show-source --statistics
         # exit-zero treats all errors as warnings. The GitHub editor is 127 chars wide
         flake8 ./python --count --exit-zero --max-complexity=10 --max-line-length=127 --statistics
-    - name: Test with pytest
+    - name: Install ngen_conf and run unit tests
       run: |
+        # setup virtual environment
+        python -m venv venv
+        source ./venv/bin/activate
+        pip install "python/ngen_conf[develop]"
         pytest python/ngen_conf
+
+        # cleanup virtual environment
+        deactivate
+        rm -rf venv
+
+    - name: Install ngen_cal and run unit tests
```

Comment:
Wouldn't this be more suited to a name indicating the tests are actually being run?

Reply:
Yeah, I think that is fair. I broke apart the workflow into more stages and changed the stage naming convention to use present tense as suggested.